### DISCUSSION TITLE
IPFS-CHAT on Android via Termux?

### DISCUSSION THREAD

**Discussion by ludetie** [SomajitDey/ipfs-chat#21] [Ideas]
Hello Somajit, 

It will be great to use IPFS-CHAT in Termux !!

Is it possible ?!

Regards,

**Comment by SomajitDey** [comment]
@ludetie Welcome back 😊

With an updated Termux it should be entirely possible. I couldn't try it as I don't have an up to date Android but I find no reason why it can't be done. Just need to install the dependencies though - viz. go-ipfs (use v0.10.0, it gives better performance); dialog; curl; openssl; tput; jq; argon2 etc.

Try it and let us know. Best wishes 👍

**Comment by ludetie** [comment]
@SomajitDey 

I installed all dependencies but it failed to launch IPFS daemon:
![image](https://user-images.githubusercontent.com/91374123/141689589-5424de5c-5a27-48d0-a93f-458c5ddac483.png)
the go-ipfs version is 0.10.0
![image](https://user-images.githubusercontent.com/91374123/141689713-03e7fd71-4f6d-4b9d-810d-f0410aafb960.png)

Any idea ? :-)

**Reply by SomajitDey** [reply]
@ludetie 

That's strange. Please tell me what happens when you launch just the ipfs daemon: 
`ipfs daemon --enable-pubsub-experiment --enable-namesys-pubsub`

**Reply by ludetie** [reply]
@SomajitDey 

![image](https://user-images.githubusercontent.com/91374123/141691691-c6034efd-0859-467f-a114-3c2b20b1d325.png)

**Reply by SomajitDey** [reply]
@ludetie IPFS seems alright. Can you check if these 3 commands are installed : `dd`, `xxd` and `nc`. Also post the screenshot of `nc -h`. If the problem is related to any of these I can bypass that.

Oh and are `flock`, `grep` and `tail` installed. They should be...but still check just to be sure.

**Comment by ludetie** [comment]
@SomajitDey 

dd, xdd, nc, flock, grep and tail are now installed, and now it starts:
From the termux session to the Ubnutu session, it can send a message to the room and also to the PM **BUT**  when sending a file to the room or to a PM it always fails:
![image](https://user-images.githubusercontent.com/91374123/141820699-5d0a2e2d-e9bc-4edc-972c-9bfcd265f177.png)
From the Ubuntu session to the termux session, it works perfectly, it can send a message to the room, to the PM and can also send a file without issue:
![image](https://user-images.githubusercontent.com/91374123/141822410-931f5f3e-143f-4c7c-8c9e-7855c07a0f66.png)


nc -h:
![image](https://user-images.githubusercontent.com/91374123/141821871-cec65567-0537-4ffb-9595-363aa058e5ea.png)

**Reply by SomajitDey** [reply]
@ludetie Great job mate! In a future release I'd remove dd, xdd etc. to reduce the dependencies. Regarding the remaining issue, plz make sure `gpg` and `openssl` are installed in Termux. Also check if `tee`, `wc` and `cut` are installed.
